### ISSUE TITLE
Back propogation 'nan' for self.weight and thus loss

### ISSUE THREAD

**Issue by ZHANG-GuiGui** [seongjunyun/Graph_Transformer_Networks#20]
Hello, Thanks for the release of your great work!
I have met with a loss 'nan' problem when I applied your GTN model on my own dataset. I have preprocessed the data following your ACM preprocessing example. However, the loss became 'nan' for the first valid process.
I have printed the tensors and I found that after the first train backpropagation, the 'self.weight' became all 'nan'. Thus, all the tensors calculated after are all 'nan'.
![image](https://user-images.githubusercontent.com/68405256/88270132-c0b87280-cd07-11ea-9c6d-33b733cc1c65.png)
I have tried smaller lr, norm=false, and modified my dataset for several different types but nothing changed.
I'd like to know if you have any idea about this problem. Thank you a lot.

**Comment by LittileFatNoRun** [comment]
I have met the same problem.

**Comment by viet98lx** [comment]
i think you can change weights initation in model.py: "self.weight = nn.Parameter(torch.Tensor(......))" to " self.weight = nn.Parameter(torch.FloatTensor(......))"

**Comment by rahzaazhar** [comment]
i got the same error

**Comment by ZHANG-GuiGui** [comment]
I haven't solved this problem. But I found that the `main_sparse.py` does not have this error.
As for the `main_sparse.py`, I've met another problem mentioned in the [issue](https://github.com/seongjunyun/Graph_Transformer_Networks/issues/23).

**Comment by ZHANG-GuiGui** [comment]
I found that the problem is in **the `norm` function** in the file `model.py`.
You can simply **remove the utilization of `normalization` function** to avoid the 'nan' error.
If you really want to do normalization for every `H`, I think that you can rewrite the `norm` function yourself.
By the way, the norm arg parser in `main.py` is inutile. The model always performs the normalization step.
